src/main/js/helpers/legend.js

```diff
@@ -44,6 +44,14 @@ const validateLegendLabel = (label) => {
  * @returns {string} Sanitized text
  */
 const getText = (text) => utils.sanitize(text);
+/**
+ * Hide legend when legend item has no data and showElement is set to false
+ *
+ * @private
+ * @param {object} input item object processed from the input JSON
+ * @returns {string} returns "none" if legend is to be hidden otherwise returns empty string 
+ */
+const hideLegend = (input) => input.legendOptions && input.legendOptions.showElement === false && utils.isEmptyArray(input.values)? "none": "";
```

Comment:
So what would happen if user-provided data for legend and `showElement` to false?

Reply:
if user-provided data for legend and showElement to false, then the legend will show up in the graph as usual. Legend will be hidden when there is no data.

Reply:
Have you verified the above condition?

Looks like `hideLegend` function will return "none"

- if `showElement` is false right? Then how will legend will be shown?

Reply:
Yes, I did unscripted testing and also written unit test to cover the same.

If ```showElement``` is false and the legend had data, then ```utils.isEmptyArray(input.values)``` will be false, as a result it will return empty string.